src/server/db/actions/GameAction.ts

```diff
@@ -72,70 +72,124 @@ export async function editGame(allData: nextEditGame) {
+      ); //Using non-null assertion, as if condition should ensure data.tags is non-null
+    }
+  }
+
+  const result = await GameModel.findByIdAndUpdate(allData.id, allData.data, {
+    new: false,
+  });
+  if (!result) {
+    throw new GenericGameErrorException("Game with given ID does not exist.");
+  }
+  return result;
+}
+export async function getSelectedGames(
+  query: z.infer<typeof GetGameQuerySchema>,
+) {
+  await connectMongoDB();
+  //Want to be able to filter by specific tags,game_builds, etc.
+  //Find the corresponding id to the Theme tag.
+  let filters: FilterQuery<IGame> = {}; //There should be a better way to build the query gradually
+  if (query.theme) {
+    //Find corresponding themeObject in mongodb
+    const found_theme: (ITheme & { id: ObjectId }) | null =
+      await ThemeModel.findOne({
+        name: query.theme,
       });
+    if (!found_theme) {
+      //No theme was found
+      throw new GenericGameErrorException(
+        `No theme with the name ${query.theme} exists.`,
+      );
     }
-  } catch (e) {
-    throw e;
+    //Add to filters object
+    filters.themes = { $all: [found_theme] };
+    //Essentially ensures that the themes array contains found_theme.
   }
-  try {
-    const result = await GameModel.findByIdAndUpdate(allData.id, allData.data, {
-      new: false,
+  //Find the correspoding tag ids for the accessibility TAGS.
+  let accessibility: ObjectId[] = [];
+  if (query.accessibility) {
+    const accessibilityTags = await TagModel.find({
+      name: { $in: query.accessibility },
+      type: "accessibility",
     });
-    if (!result) {
-      throw new GenericUserErrorException("Game with given ID does not exist.");
+    //Ensure that we found all tags and no extras
+    if (query.accessibility.length !== accessibilityTags.length) {
+      throw new GenericGameErrorException(
+        "One or more of the accessibility tags do not exist.",
+      );
     }
-    return result;
-  } catch (e) {
-    throw e;
+    accessibility = accessibilityTags.map((tag) => tag.id);
   }
-}
+  let custom: ObjectId[] = [];
+  if (query.tags) {
+    const normalTags = await TagModel.find({
+      name: { $in: query.tags },
+      type: "custom",
+    });
+    if (query.tags.length !== normalTags.length) {
+      throw new GenericGameErrorException(
+        "One or more of the custom tags do not exist.",
+      );
+    }
+    custom = normalTags.map((tag) => tag.id);
+  }
+  const totalTags = [...custom, ...accessibility];
+  if (totalTags.length !== 0) {
+    //Add to filters
+    filters.tags = { $all: totalTags };
+  }
+  //Currently ignoring filtering by build, as that hasn't yet been implemented.
 
-export async function getAllGames() {
-  await connectMongoDB();
-  try {
-    const games = await GameModel.find();
-    if (games == null) {
-      return [];
+  //Match by name with regex
+  if (query.name && query.name !== "") {
+    const reg_string = new RegExp(query.name, "i");
```

Comment:
This should match partial matches as well - not just exact matches, use `*` to match 0 or more characters in both directions